repo_health/check_dependabot_alerts.py

```diff
@@ -0,0 +1,78 @@
+"""
+Checks repository open dependabot alert and collects metrics.
+"""
+
+import json
+import logging
+import os
+
+import requests
+from pytest_repo_health import add_key_to_metadata
+
+from .utils import github_org_repo
+
+logger = logging.getLogger(__name__)
+
+MODULE_DICT_KEY = "dependabot_alerts"
+
+def get_github_dependabot_api_response(org_name, repo_name):
+    """
+    get the repos dependabot information using api
+    """
+
+    response = requests.get(
+        url=f'https://api.github.com/repos/{org_name}/{repo_name}/dependabot/alerts',
+        headers={'Authorization': f'Bearer {os.environ["GITHUB_TOKEN"]}'}
+    )
+
+    if response.status_code != 200:
+        logger.error(
+                "An error occurred while fetching %s. status code %s content info %s.",
+                repo_name,
+                response.status_code,
+                response.content
+        )
+        return None
+
+    return response.json()
+
+@add_key_to_metadata((MODULE_DICT_KEY, ))
```

Comment:
We'll want these under an intermediate key, perhaps `dependabot_alerts`. I believe this is the correct way to include documentation for the leaf keys at the same time:
```suggestion
@health_metadata(
    [MODULE_DICT_KEY, "dependabot_alerts"],
    {
        "total_count": "The total number of open Dependabot alerts for this repository",
        "low_severity ": "Open Dependabot alerts with low severity",
        "medium_severity ": "Open Dependabot alerts with medium severity",
        "high_severity ": "Open Dependabot alerts with high severity",
        "critical_severity ": "Open Dependabot alerts with critical severity",
    }
)
```

Reply:
the only reason why i'd push back on this is because if i were to use https://github.com/openedx/edx-repo-health/blob/master/repo_health/check_github.py#L130 as an example, then i'd expect a large chunk of extra columns in the dashboard sheet. However, I don't see them present, so before i commit this, i'd want to verify where this'll go in the sheet.

Reply:
The checks in that file currently don't work due to a bug with async checks, which is why you don't see them. But that sheet gets recreated every time the job runs, so any new keys will show up as soon as they start getting produced.